python_modules/dagster/dagster/_core/definitions/selector.py

```diff
@@ -60,6 +62,9 @@ def with_solid_selection(self, solid_selection):
             self.location_name, self.repository_name, self.pipeline_name, solid_selection
         )
 
+    def get_id(self) -> str:
+        return hash_str(json.dumps(self._asdict()))
```

Comment:
@alangenfeld this is inspired from serdes (this is not officially a serdes class) - should be consistent though? Unless serdes does some additional ordering more than json.dumps does

Reply:
that should be fine but some relevant code in https://github.com/dagster-io/dagster/pull/12293/files if you defer til thats landed

Reply:
What would be the drawbacks of naming this `__hash__` and use these objects directly as the dictionary key?